Bug 1068691 - set aria role and label for close

jobara opened the pull request:
Set the aria role (button) and label for the close button on the message-list edit view.

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/556024?v=3" alt="jobara" width="50px" height="50px" /> Justin Obara (jobara)](https://github.com/jobara) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=313498eda83e)

yzen commented on apps/email/js/cards/message_list.html:
```diff
@@ -33,8 +33,9 @@
-       class="msg-listedit-cancel-btn">
-      <span class="icon icon-close"></span>
+       class="msg-listedit-cancel-btn" role="button">
+      <span class="icon icon-close" data-l10n-id="message-listEdit-close">
```
Put a label on the anchor (button) itself instead. Also change the name to 'close-button' as it is pretty general and we might have to reuse it elsewhere.

yzen commented on apps/email/locales/email.en-US.properties:
```diff
@@ -673,6 +673,11 @@ message-list-top-action=Top
+# LOCALIZATION NOTE(message-listEdit-close): For the message-list edit view,
+# a label for a header icon button that closes the message-list edit view.
+# The following string is spoken by screen readers and not shown on the screen.
+message-listEdit-close.ariaLabel=Close
```
See above, change the name to close-button and also move it to the top of the file besides back-button entry.

jobara committed:
Bug 1068691 - set aria role and label for close

Set the aria role (button) and label for the close button on the message-list edit view.

jobara commented:
@yzen thanks for the review. I've made the changes you recommended.

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/556024?v=3" alt="jobara" width="50px" height="50px" /> Justin Obara (jobara)](https://github.com/jobara) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f8e7c77034ec)
